source/common/network/cidr_range.cc

```diff
@@ -79,13 +79,20 @@ IpVersion CidrRange::version() const {
 }
 
 bool CidrRange::isInRange(InstanceConstSharedPtr address) const {
-  if (address == nullptr || length_ < 0 || address->type() != Type::Ip ||
-      address->ip()->version() != version()) {
+  if (address == nullptr) {
+    return false;
+  }
+
+  return isInRange(*address);
+}
+
+bool CidrRange::isInRange(const Instance& address) const {
+  if (length_ < 0 || address.type() != Type::Ip || address.ip()->version() != version()) {
     return false;
   }
   // Make an CidrRange from the address, of the same length as this. If the two ranges have
   // are the same, then the address is in this range.
-  CidrRange other = create(address, length_);
+  CidrRange other = create(address.ip()->addressAsString(), length_);
```

Comment:
If you're not going to make the change in this PR, can you add a `TODO(hennna): ...` here? Personally I think it's crazy that we are munging back and forth between strings to do what are simple bitmask operations at the end of the day, but I don't feel so strongly I'll block the merge on this if a TODO is added.